### PR TITLE
Option to specify an absolute salt:// path for config_files

### PR DESCRIPTION
**Pull request by daks**
### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [X] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [X] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

This solves #56 

### Describe the changes you're proposing

The proposed changes lets you define a config file, using `config_files` pillar in two ways:
* the previous one which search in `slspath`
* the new one anywhere in `salt://` fileserver. Just prefix it with `salt://` to use this one

### Pillar / config required to test the proposed changes

Please see `pillar.example` which is used for the additional test.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [X] Updated `pillar.example`.

### Testing checklist

- [X] Included in Kitchen (i.e. under `state_top`).
- [X] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [X] Updated the relevant test pillar.

### CODE REVIEW

**Comment by daks** [comment]
yamllint raises an error 

```
pillar.example
  44:1      warning  comment not indented like content  (comments-indentation)
```
not sure how to solve it. And Debian 10 fails on verify, I imagine this is the same `train` lib problem found by @myii

File: pillar.example [hunk 1]
```diff
@@ -39,12 +39,14 @@ rabbitmq:
   config_files:
     # config_files, if you need some stuffs set directly in configuration
     # files, rather than via the light modules available.
-    rabbitmq.config:
-      # Need to add some template in rabbitmq/templates/rabbitmq.config
-      # templates not provided, as is probably too specific...
-      source: templates/relay-rabbitmq.config
+    rabbitmq.conf:
+      # source is relative to `rabbitmq` formula
+#      source: templates/rabbitmq.conf
```
Comment (myii) [inline pillar.example]: The `comments-indentation` error usually requires the indentation of the comment character to be in line with _the line below_, so in this case:

```suggestion
      # source: templates/rabbitmq.conf
```

**Comment by myii** [comment]
> ... And Debian 10 fails on verify, I imagine this is the same `train` lib problem ...

@daks That commit has already been added to this formula.  Actually, there's failures during the verification:

* https://travis-ci.com/saltstack-formulas/rabbitmq-formula/jobs/268517206#L1777-L1783

```sls
  ×  rabbitmq configuration: should match desired lines (3 failed)
     ×  File /etc/rabbitmq/rabbitmq.conf is expected to exist
     expected File /etc/rabbitmq/rabbitmq.conf to exist
     ×  File /etc/rabbitmq/rabbitmq.conf is expected to be file
     expected `File /etc/rabbitmq/rabbitmq.conf.file?` to return true, got false
     ×  File /etc/rabbitmq/rabbitmq.conf content is expected to include "# Config file for rabbitmq"
     expected nil to include "# Config file for rabbitmq", but it does not respond to `include?`
```

File: rabbitmq/config_files.sls [hunk 2]
```diff
@@ -2,10 +2,15 @@
...
+{%- for filename, info in salt["pillar.get"]("rabbitmq:config_files", {}).items() %}
+  {%- set source = info['source'] %}
 /etc/rabbitmq/{{ filename }}:
   file.managed:
-    - source: salt://{{ slspath }}/{{ info['source'] }}
+    {%- if source.startswith('salt://') %}
+    - source: {{ source }}
+    {%- else %}
+    - source: salt://{{ slspath }}/{{ source }}
+    {% endif %}
```
Comment (myii) [inline rabbitmq/config_files.sls]: Should we squash this into a one-liner?

**Comment by myii** [comment]
@daks Rather than have this as a separate state (i.e. in Kitchen), shouldn't `.config_files` be added to this section?

https://github.com/saltstack-formulas/rabbitmq-formula/blob/e97c976c4b3f3c38ff05886787289ca191912e73/rabbitmq/install.sls#L3-L9

**Comment by daks** [comment]
> @daks Rather than have this as a separate state (i.e. in Kitchen), shouldn't `.config_files` be added to this section?
> 
> https://github.com/saltstack-formulas/rabbitmq-formula/blob/e97c976c4b3f3c38ff05886787289ca191912e73/rabbitmq/install.sls#L3-L9
> 
> Or does it have any relationship to [`rabbitmq/config.sls`](https://github.com/saltstack-formulas/rabbitmq-formula/blob/master/rabbitmq/config.sls)?

I don't know. I wanted a minimal PR with no possible influence on existing behaviour.

**Comment by myii** [comment]
@daks Thanks, I've merged this after our conversation in Slack.  Going forward, `config_files` should be added to the `README`.

**Comment by daks** [comment]
thanks

**Pull request merged**